types.ts

```diff
@@ -152,3 +152,20 @@ export interface CircleProps {
 
 export type ColorScale = d3.ScaleOrdinal<string, string, never>
 
+export type ScaleFunc =
+  | d3.ScaleLinear<number, number, never>
+  | d3.ScaleTime<number, number, never>
+
+export type AccessorFunc = (d: any) => number | Date;
```

Comment:
do we want to make d of type Data?

Reply:
Only Date or number, same one applies to x and y accessors and x may be date or number while y is only number